ext/proj4_c_impl/main.c

```diff
@@ -301,6 +301,58 @@ static VALUE method_proj4_auth_name_str(VALUE self)
   return result;
 }
 
+static VALUE method_proj4_axis_and_unit_info_str(VALUE self, VALUE dimension)
+{
+  VALUE result;
+  int dimension_index;
+  PJ *pj;
+  PJ *pj_cs;
+  const char *axis_info;
+  const char *unit_name;
+  RGeo_Proj4Data *data;
+
+  Check_Type(dimension, T_FIXNUM);
+
+  dimension_index = FIX2INT(dimension);
+  result = Qnil;
+
+  TypedData_Get_Struct(self, RGeo_Proj4Data, &rgeo_proj4_data_type, data);
+  pj = data->pj;
+  if (pj){
+    pj_cs = proj_crs_get_coordinate_system(PJ_DEFAULT_CTX, pj);
+    if (pj_cs && proj_cs_get_axis_info(PJ_DEFAULT_CTX, pj_cs, dimension_index, &axis_info, NULL, NULL, NULL, &unit_name, NULL, NULL)) {
+      result = rb_sprintf("%s:%s", axis_info, unit_name);
+
+      proj_destroy(pj_cs);
```

Comment:
you 
may not destroy this object in some branches (the object is present but `proj_cs_get_axis_info` fails)

Reply:
Fixed. I broke it out into a nested if statement. In my testing, `proj_destroy` accepts `NULL` and doesn't seem to complain, so we could probably just move `proj_destroy` out of this block, but I'm not confident that is true for all versions and this is a little more explicit.